src/configuration/utils.ts

```diff
@@ -0,0 +1,15 @@
+import { PermissionEnum } from "@saleor/types/globalTypes";
+import { IntlShape } from "react-intl";
+
+import { createConfigurationMenu } from ".";
+
+export const getConfigMenuItemsPermissions = (
+  intl: IntlShape
+): PermissionEnum[] =>
+  createConfigurationMenu(intl)
+    .reduce(
+      (prev, next) =>
+        prev.concat(next.menuItems.map(menuItem => menuItem.permissions)),
+      []
+    )
+    .reduce((prev, next) => prev.concat(next), []);
```

Comment:
There's a built-in method for that called `.flat()`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat